### ISSUE TITLE
[BUG] Prettier class order don't match with eslint-plugin-tailwindcss

### ISSUE THREAD

**Issue by snikoletopoulos** [francoismassart/eslint-plugin-tailwindcss#231]
**Describe the bug**
With the prettier and eslint plugin installed in some cases the class order don't match.

**To Reproduce**
In a div element add the classes `top-0 left-0`
prettier will not change the class orders but there will be an eslint error
applying eslint fix will satisfy both eslint and prettier plugin the class names should now be `left-0 top-0`

**Expected behavior**
when I format with prettier, this error should go away

**Screenshots**
With prettier formating:
![image](https://user-images.githubusercontent.com/78855607/229350059-ded77e2f-4785-47d3-ac49-1c054f403af6.png)

With eslint fix applied (prettier is ok as well with this one):
![image](https://user-images.githubusercontent.com/78855607/229350083-de9da589-8593-456b-afec-2f17574eaad6.png)

**Comment by francoismassart** [comment]
Notes for self: `absolute left-0 top-0 transition-transform group-hover:scale-110`

**Comment by francoismassart** [comment]
@snikoletopoulos 
Using the "Tidy" button from https://play.tailwindcss.com/0Fyv3SDjWC

The order is `<div class="absolute left-0 top-0 transition-transform group-hover:scale-110"></div>`

And if I add this exact snippet in the test cases of the `classname-order` rule, it passes as valid order...

Maybe could you share a minimalistic project demonstrating your specific issue so I can debunk it. thx

**Comment by anup-a** [comment]
Getting similar issues for the following -


`focus:outline-none focus:ring-2 focus:ring-primary-500/25 focus:ring-offset-2` 

`data-[state=open]:bg-light-grey disabled:pointer-events-none`

`focus:outline-none focus-visible:ring focus-visible:ring-primary-500/25`

**Comment by nikitastryuk** [comment]
+1 
conflicting `mx-2 flex` vs `flex mx-2`

**Comment by matheins** [comment]
Having the same issue

**Comment by JaKXz** [comment]
Curious if anyone has a workaround for this? I am probably going to disable the rule for now, but I would prefer to default to eslint

**Comment by adonaicandido** [comment]
having the same issue, but when using `before:`

**Comment by birtles** [comment]
> conflicting `mx-2 flex` vs `flex mx-2`

Similarly `block mt-8` (eslint-plugin-tailwindcss) vs `mt-8 block` (prettier)

**Comment by pdlug** [comment]
I'm seeing the same issue w/ brand colors (ex: `border border-primary-500` gets reordered to `border-primary-500 ... border` by `eslint-plugin-react` but preserved as original by `prettier-plugin-tailwindcss`.

**Comment by haikbvn** [comment]
I have similar issue:

**prettier-plugin-tailwindcss**
```typescript
<header className="sticky top-0 z-40 w-full border-b bg-background">
```

**eslint-plugin-tailwindcss**
```typescript
<header className="bg-background sticky top-0 z-40 w-full border-b">
```

**Comment by ImamAlfariziSyahputra** [comment]
> Here it conflicts both ways, brandPrimary is a custom color
> 
> prettier `outline-brandPrimary p-2`
> 
> eslint-config-tailwindcss `p-2 outline-brandPrimary`
> 
> I'll look if i can find an example with no custom themes

i'm facing the same issue..

**Comment by Ouckah** [comment]
Having a similar issue:

On save: `min-w-96 flex h-full w-full`

What ESLint wants: `flex h-full w-full min-w-96`

Any way to fix this?

**Comment by ryanlonstein** [comment]
I am having the same issue.

Prettier wants:
```
size-full absolute
```
ESLint wants:
```
absolute size-full
```

**Comment by scr2em** [comment]
any updates ?

**Comment by liamlows** [comment]
Is there any solution to this other than removing the `prettier-plugin-tailwindcss` plugin from the `.prettierrc` file? 

Similar to what others have said when I have `eslint-plugin-tailwindcss` installed and 
```extends: ['plugin:tailwindcss/recommended', ...]``` 
set up in the `.eslintrc.js` file and also have `prettier-plugin-tailwindcss` in the plugins array in `.prettierrc` and linter wants it ordered one way and prettier wants it another just like @ryanlonstein said.

**Comment by kachkaev** [comment]
There is no need for a Prettier plugin if this ESLint plugin is installed. But if you want to use both, you can disable a ESLint rule that sorts class names:

```js
// .eslintrc
module.exports = {
  extends: [
    "...",
    "plugin:tailwindcss/recommended",
  ],
  rules: {
    "...": "...",
    "tailwindcss/classnames-order": "off"
  }
};
```

**Comment by liamlows** [comment]
> There is no need for a Prettier plugin if this ESLint plugin is installed. But if you still want to use both, you can disable a ESLint rule that sorts class names:
> 
> ```js
> // .eslintrc
> module.exports = {
>   extends: [
>     "...",
>     "plugin:tailwindcss/recommended",
>   ],
>   rules: {
>     "...": "...",
>     "tailwindcss/classnames-order": "off"
>   }
> };
> ```
> 
> Note that what [`prettier-plugin-tailwindcss`](https://www.npmjs.com/package/prettier-plugin-tailwindcss) does is [outside the scope of Prettier](https://prettier.io/docs/en/rationale.html#what-prettier-is-_not_-concerned-about):
> 
> > Prettier only _prints_ code. It does not transform it. This is to limit the scope of Prettier. Let’s focus on the printing and do it really well!
> > Here are a few examples of things that are out of scope for Prettier:
> > 
> > * Turning single- or double-quoted strings into template literals or vice versa.
> > * Using `+` to break long string literals into parts that fit the print width.
> > * Adding/removing `{}` and `return` where they are optional.
> > * Turning `?:` into `if`-`else` statements.
> > * Sorting/moving imports, object keys, class members, JSX keys, CSS properties or anything else. Apart from being a _transform_ rather than just printing (as mentioned above), sorting is potentially unsafe because of side effects (for imports, as an example) and makes it difficult to verify the most important [correctness](https://prettier.io/docs/en/rationale.html#correctness) goal.

Gotcha, I figured that was the case but wasn't for sure. It might be helpful to add a note to the readme when talking about installation that the prettier plugin and this plugin are not complimentary (or at least meant to be) to avoid confusion. Thanks for the quick response btw!

**Comment by mrleblanc101** [comment]
Prettier: 
`class="h-14 w-full rounded-md border-white bg-tropical-50 bg-[url('/img/calendar.svg')] bg-[length:1.5rem] bg-[position:center_left_0.75rem] bg-no-repeat pl-12 text-left focus:border-tropical-50 focus:bg-white focus:ring-2 focus:ring-tropical-300 lg:w-1/4 lg:max-w-[252px]"`
Eslint:
`class="bg-tropical-50 focus:border-tropical-50 focus:ring-tropical-300 h-14 w-full rounded-md border-white bg-[url('/img/calendar.svg')] bg-[length:1.5rem] bg-[position:center_left_0.75rem] bg-no-repeat pl-12 text-left focus:bg-white focus:ring-2 lg:w-1/4 lg:max-w-[252px]"`

**Comment by ItsTarik** [comment]
@kachkaev but this will disable the error within eslint ? so prettier will reorder without having a clear reson ?

**Comment by sjdonado** [comment]
isn't there a way to disable this rule in prettier instead? this is very annoying

**Comment by francoismassart** [comment]
Fixed in v4, it will be release after cleaning up the existing issues.
